test/libethereum/test/boostTest.cpp

```diff
@@ -117,5 +127,10 @@ int main( int argc, char* argv[] )
 	for (int i = 0; i < argc; i++)
 		parameters.push_back(argv[i]);
 
-	return unit_test_main(fake_init_func, argc, argv);
+	std::future<int> ret = std::async(unit_test_main, fake_init_func, argc, argv);
+	std::thread outputThread(travisOut);
+	int result = ret.get();
+	stopTravisOut = true;
```

Comment:
Whatever you do in separated threads you must sync memory writes/reads. In your case change to stopTravisOut may be never visible to other thread (usually not the case on x86 architecture).